<a href="https://colab.research.google.com/github/kelvindules/pucrio-mvp4/blob/main/colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, classification_report
from xgboost import XGBClassifier

In [2]:
# Carregando dados
csv_url = 'https://raw.githubusercontent.com/kelvindules/pucrio-mvp4/main/tests/game_teams_stats.csv'
df = pd.read_csv(csv_url)

In [3]:
# Removendo registros ruins
df = df.dropna()

In [4]:
# Indicadores de jogadas agressivas
X = df[["hits", "takeaways", "faceOffWinPercentage"]]

# Indicador de vitória
y = df["won"]

In [5]:
# Dividindo o dataset em treino e teste
X_train, X_test, y_train, y_test = train_test_split(
  X, y, test_size=0.2, random_state=42
)

In [6]:
# DecisionTreeClassifier
dt_classifier = DecisionTreeClassifier(random_state=42)
dt_classifier.fit(X_train, y_train)
dt_prediction = dt_classifier.predict(X_test)

# RandomForestClassifier
rf_classifier = RandomForestClassifier(random_state=42)
rf_classifier.fit(X_train, y_train)
rf_prediction = rf_classifier.predict(X_test)

# XGBClassifier
xgb_classifier = XGBClassifier(random_state=42)
xgb_classifier.fit(X_train, y_train)
xgb_prediction = xgb_classifier.predict(X_test)


In [7]:
print("DecisionTreeClassifier:")
print("Precisão:", accuracy_score(y_test, dt_prediction))
print("Relatório:\n", classification_report(y_test, dt_prediction))

print("RandomForestClassifier:")
print("Precisão:", accuracy_score(y_test, rf_prediction))
print("Relatório:\n", classification_report(y_test, rf_prediction))

print("XBGClassifier:")
print("Precisão:", accuracy_score(y_test, xgb_prediction))
print("Relatório:\n", classification_report(y_test, xgb_prediction))

DecisionTreeClassifier:
Precisão: 0.62003045170022
Relatório:
               precision    recall  f1-score   support

       False       0.61      0.66      0.63      2912
        True       0.64      0.58      0.61      2999

    accuracy                           0.62      5911
   macro avg       0.62      0.62      0.62      5911
weighted avg       0.62      0.62      0.62      5911

RandomForestClassifier:
Precisão: 0.6163085772288953
Relatório:
               precision    recall  f1-score   support

       False       0.61      0.62      0.61      2912
        True       0.62      0.62      0.62      2999

    accuracy                           0.62      5911
   macro avg       0.62      0.62      0.62      5911
weighted avg       0.62      0.62      0.62      5911

XBGClassifier:
Precisão: 0.564879039079682
Relatório:
               precision    recall  f1-score   support

       False       0.56      0.57      0.56      2912
        True       0.57      0.56      0.57      2999


Os resultados demonstram que há certa correlatividade entre as jogadas agressivas (hits, takeaways e faceoffs) e os resultados das partidas. Todos os modelos selecionados alcançam precisão maior que 0.55, sendo o melhor para o nosso caso de uso o modelo de Árvore de Decisão, que alcança 0.62 no dataset utilizado.